src/components/SmartContractJsonSchema/renderPrimitivesType.tsx

```diff
@@ -0,0 +1,281 @@
+
+import { validate } from "@/validate";
+
+import { PositiveIntPicker } from "@/components/FormElements/PositiveIntPicker";
+
+import type { AnyObject, SorobanInvokeValue } from "@/types/types";
+import { convertSpecTypeToScValType } from "@/helpers/sorobanUtils";
+
+export const renderPrimitivesType = ({
+  name,
+  schema,
+  path,
+  parsedSorobanOperation,
+  onChange,
+  formError,
+  setFormError,
+}: {
+  name: string;
+  schema: Partial<JSONSchema7>;
+  path: string[];
+  parsedSorobanOperation: SorobanInvokeValue;
+  onChange: (value: SorobanInvokeValue) => void;
+  formError: AnyObject;
+  setFormError: (error: AnyObject) => void;
+}) => {
+  const { description } = schema;
+
+  const schemaType = jsonSchema.getSchemaType(schema);
+
+  const nestedItemLabel =
+    path.length > 0 ? jsonSchema.getNestedItemLabel(path.join(".")) : path;
+
+  const formErrorKey = [
+    parsedSorobanOperation.function_name,
+    path.join("."),
+  ].join(".");
+
+  const sharedProps = {
+    id: path.join("."),
+    label: `${nestedItemLabel} (${schemaType})`,
+    value: get(parsedSorobanOperation.args, path.join("."))?.value || "",
+    error: formError?.[formErrorKey] || undefined,
+  };
+
+  const invokeContractBaseProps = {
+    contract_id: parsedSorobanOperation.contract_id,
+    function_name: parsedSorobanOperation.function_name,
+  };
+
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+    schemaType: string,
+  ) => {
+    const scValType = convertSpecTypeToScValType(schemaType);
+
+    if (path.length > 0) {
+      const updatedList = jsonSchema.setDeepValue(
+        parsedSorobanOperation.args,
+        path.join("."),
+        {
+          value: e.target.value,
+          type: scValType,
+        },
+      );
+
+      onChange({
+        ...invokeContractBaseProps,
+        args: {
+          ...updatedList,
+        },
+      });
+    } else {
+      onChange({
+        ...invokeContractBaseProps,
+        args: {
+          ...parsedSorobanOperation.args,
+          [name]: { value: e.target.value, type: scValType },
+        },
+      });
+    }
+  };
+
+  const handleValidate = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    schemaType: string,
+    validateFn?:
+      | ((value: string, required?: boolean) => string | false)
+      | ((value: string, required?: boolean) => string | false)[],
+  ) => {
+    let error;
+
+    if (Array.isArray(validateFn)) {
+      const errors = validateFn.map((fn) => fn(e.target.value));
+      const hasNoError = jsonSchema.hasAnyValidationPassed(errors);
+
+      if (schemaType === "Address") {
+        error = hasNoError ? "" : "Invalid Public key or contract ID";
+      } else {
+        error = hasNoError ? "" : errors.join(" ");
+      }
+    } else {
+      error = validateFn?.(e.target.value);
+    }
+
+    if (error) {
+      setFormError({
+        ...formError,
+        [formErrorKey]: error ? error : "",
```

Comment:
We should always have truthy `error` here, right?

```suggestion
        [formErrorKey]: error,
```